indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -374,6 +406,20 @@
         (index-concept context concept parsed-concept options)
         (log-ingest-to-index-time concept)))))
 
+(defn- cascade-collection-delete
+  "Performs the cascade actions of collection deletion,
+  i.e. propagate collection deletion to granules and variables"
+  [context concept-mapping-types concept-id revision-id]
+  (doseq [index (idx-set/get-granule-index-names-for-collection context concept-id)]
```

Comment:
I've seen this fail on occasion when there are a lot of granules - if an exception is thrown we won't reindex the associated variables. In this case the variable associations would not get updated - what do you think about putting a message on the queue to index the variable instead and doing that part first?

We don't have to hold up the pull request for this, just something to think about.